### PR TITLE
feat(blob.py): auto-populate standard headers for non-chunked downloads

### LINKED ISSUE #24: Storage: Capture relevant headers to blob properties during download
Residual from googleapis/google-cloud-python#9003.

@william-silversmith notes that even with `raw_download` enabled, he is unable to detect the `content_type` of a downloaded blob without performing an additional `reload` request, which is prohibitive for his usecase at scale.  E.g.:

```python
blob = bucket.blob( key )
binary = blob.download_as_string(raw_download=True)
if blob.content_encoding == 'gzip':
    return gunzip(binary)
elif blob.content_encoding == 'br':
    return brotli.decompress(binary)
else:
    return binary
```
Potentially even...

```python
if blob.content_type == 'application/json':
    return json.loads(binary.decode('utf8'))
```

### PR DESCRIPTION
**Pull request by william-silversmith**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #24  🦕

This PR autopopulates the following fields for non-chunked downloads based on the server header response:
```
blob.content_encoding
blob.content_type
blob.cache_control
blob.storage_class
blob.content_language
blob.md5_hash
blob.crc32c
```

### CODE REVIEW

**Comment by william-silversmith** [comment]
@tseaver would these unit tests be sufficient? I can get more involved if necessary. I had trouble invoking _do_download, it was interacting strangely with "MagicMock" and dying on closing a file stream.

**Comment by tseaver** [comment]
@william-silversmith Thanks, that new test was sufficient.

**Comment by william-silversmith** [comment]
I'm a little confused because 806-808 have to be evaluated for the crc32c and md5_hash tests to pass.

File: google/cloud/storage/blob.py [hunk 1]
```diff
@@ -783,6 +784,34 @@ def _get_download_url(
...
+            :class requests.models.Response
+        :param response: The server response from downloading a non-chunked file
+        """
+        self.content_encoding = response.headers.get("Content-Encoding", None)
+        self.content_type = response.headers.get("Content-Type", None)
+        self.cache_control = response.headers.get("Cache-Control", None)
+        self.storage_class = response.headers.get("X-Goog-Storage-Class", None)
+        self.content_language = response.headers.get("Content-Language", None)
+        #  'X-Goog-Hash': 'crc32c=4gcgLQ==,md5=CS9tHYTtyFntzj7B9nkkJQ==',
+        x_goog_hash = response.headers.get("X-Goog-Hash", '')
```
Comment (tseaver) [inline google/cloud/storage/blob.py:803]: `black` hates single quotes:
```suggestion
        x_goog_hash = response.headers.get("X-Goog-Hash", "")
```

Comment (william-silversmith) [reply google/cloud/storage/blob.py]: Thanks!

**Comment by tseaver** [comment]
@googlebot I consent

**Comment by william-silversmith** [comment]
@googlebot I fixed it.

**Comment by william-silversmith** [comment]
I could try force pushing a double quote update that removes you as an author if that would be okay with you?

**Comment by tseaver** [comment]
@william-silversmith Sure, WFM.

**Comment by tseaver** [comment]
W00t!  Thanks for persevering, @william-silversmith .

**Comment by william-silversmith** [comment]
Excited thank you!

**Pull request merged**